### PR TITLE
Add scheduled debate date to awaiting debate petitions view

### PR DESCRIPTION
**Pull request by leandroalemao**
Hi people,

I've added the debate schedule date to the awaiting debate petitions view. Useful to check the debate date before clicking on the petition.

image

### CODE REVIEW

File: app/helpers/date_time_helper.rb [hunk 1]
```diff
@@ -44,4 +44,13 @@ def api_date_format(date_time)
       end
     end
   end
+
+  def to_be_debated_on_in_words(date)
+    return unless date.present?
```
Comment (pixeltrix) [inline app/helpers/date_time_helper.rb]: By doing the checking for the debate date here you end up with and empty `<p>` tag which is undesirable - we should use `scheduled_debate_date?` on the petition in the view and remove this check. E.g:

``` ruby
<% if petition.scheduled_debate_date? %>
<p><%= to_be_debated_on_in_words(petition.scheduled_debate_date) %></p>
<% end %>
```

File: spec/helpers/date_time_helper_spec.rb [hunk 2]
```diff
@@ -108,4 +108,20 @@
       end
     end
   end
+
+  describe "#to_be_debated_on_in_words" do
+    context "when the date is nil" do
+      it "returns nil" do
+        expect(helper.to_be_debated_on_in_words(nil)).to be_nil
+      end
+    end
```
Comment (pixeltrix) [inline spec/helpers/date_time_helper_spec.rb]: Obviously we need to remove this test if we add the condition to the view.

**Comment by pixeltrix** [comment]
Other than the comments it looks 👍 

@henryhadlow do we want to do relative dates, eg. 'To be debated today', 'To be debated tomorrow' ?

**Comment by leandroalemao** [comment]
@pixeltrix tks for replying.. 

I agree.. and I'm doing the changes right now and waiting for @henryhadlow reply.

:+1:

File: app/helpers/date_time_helper.rb [hunk 3]
```diff
@@ -45,11 +46,12 @@ def api_date_format(date_time)
     end
   end
 
-  def to_be_debated_on_in_words(date)
+  def to_be_debated_on_in_words(date, today = Date.today)
```
Comment (pixeltrix) [inline app/helpers/date_time_helper.rb]: Needs to be `Date.current` to take account of application timezone

File: spec/helpers/date_time_helper_spec.rb [hunk 4]
```diff
@@ -108,4 +108,30 @@
...
+    context "when the date is today" do
+      let(:date) { Date.parse("11/11/2016") }
+
+      it "returns 'To be debated on 11 November 2016'" do
+        expect(helper.to_be_debated_on_in_words(date)).to eq("To be debated on 11 November 2016")
+      end
+    end
+
+    context "when the date is today" do
+      let(:date) { Date.today }
```
Comment (pixeltrix) [inline spec/helpers/date_time_helper_spec.rb]: As above, it needs to be `Date.current`

Comment (leandroalemao) [inline spec/helpers/date_time_helper_spec.rb]: :+1: :smile:

File: app/helpers/date_time_helper.rb [hunk 5]
```diff
@@ -44,4 +45,14 @@ def api_date_format(date_time)
       end
     end
   end
+
+  def to_be_debated_on_in_words(date, today = Date.current)
+    return unless date.present?
```
Comment (pixeltrix) [inline app/helpers/date_time_helper.rb]: You don't need this check here now since we're checking in the view for the presence of the date

Comment (leandroalemao) [inline app/helpers/date_time_helper.rb]: Yes I agree.. sorry, I forgot to remove it.. but it's all done now.. commit [4700c6f](https://github.com/alphagov/e-petitions/pull/456/commits/4700c6f01e13f3292772e6934e70973e2dac54d6) 👍 😄 tks

**Comment by pixeltrix** [comment]
@leandroalemao thanks!

**Comment by leandroalemao** [comment]
@pixeltrix, I'm glad to see you've merged my pull request.. 👍 😄 I'm willing to contribute more and more. Tks!

**Comment by pixeltrix** [comment]
@leandroalemao it was a pleasure - if you're looking for something else to work on then #450 is a nice and easy one but if you're looking for something a bit deeper then #458 should be interesting. Feel free to ask questions if you need anything clarifying.

**Pull request merged**